jobs/mongodb_migration/src/mongodb_migration/migrations/_20230309141600_cache_add_job_runner_version.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+
+from libcommon.simple_cache import CachedResponse
+from mongoengine.connection import get_db
+
+from mongodb_migration.check import check_documents
+from mongodb_migration.migration import Migration
+
+
+# connection already occurred in the main.py (caveat: we use globals)
+class MigrationAddJobRunnerVerionToCacheResponse(Migration):
+    def up(self) -> None:
+        # See https://docs.mongoengine.org/guide/migration.html#example-1-addition-of-a-field
+        logging.info("If missing, add 'job_runner_version' field based on 'worker_version' value")
+        db = get_db("cache")
+        db["cachedResponsesBlue"].update_many(
+            {"job_runner_version": {"$exists": False}},
```

Comment:
Implicitly we expect that $worker_version exists. It should, but what if it does not exist? Maybe it will break.

I worker_version does not exist, we should set a default value instead.

Also: we expect a specific format for worker_version (it should), but what if we have weird values there?

To be extra safe, maybe we could do two migrations here:
- first create a job_runner everywhere, using worker_version when possible, but also handling the case where worker_version does not exist or has not the correct format (not sure how... we could just check what is the current state of the database -> if there are no such cases, maybe just skip this part)
- in the same PR: only use job_runner_version in the code of datasets-server
- then, in a following PR, delete worker_version.

WDYT?

Reply:
Makes sense, it could be much better